aiohttp/connector.py

```diff
@@ -69,6 +72,39 @@
     from .tracing import Trace
 
 
+def _convert_hosts_to_addr_infos(
+    hosts: List[Dict[str, Any]]
+) -> List[aiohappyeyeballs.AddrInfoType]:
+    """Converts the list of hosts to a list of addr_infos.
+
+    The list of hosts is the result of a DNS lookup. The list of
+    addr_infos is the result of a call to `socket.getaddrinfo()`.
+    """
+    addr_infos: List[aiohappyeyeballs.AddrInfoType] = []
+    for hinfo in hosts:
+        host = hinfo["host"]
+        is_ipv6 = helpers.is_ipv6_address(host)
+        family = socket.AF_INET6 if is_ipv6 else socket.AF_INET
+        addr = (host, hinfo["port"], 0, 0) if is_ipv6 else (host, hinfo["port"])
+        addr_infos.append(
+            (family, socket.SOCK_STREAM, socket.IPPROTO_TCP, hinfo["hostname"], addr)
+        )
+    return addr_infos
```

Comment:
We could store the addresses in the cache in python's `getaddrinfo` format to avoid having to convert.  It would be more efficient and more portable.... but its a bit less readable.  Something to consider for the future though